### ISSUE TITLE
Yolov8 finetune issue

### ISSUE THREAD

**Issue by tzktok** [ultralytics/ultralytics#21510]
# YOLOv8 Fine-Tuning Performance Drop Issue

## Overview

This repository documents a critical issue encountered while fine-tuning a high-performing YOLOv8 model on a smaller dataset. Despite strong initial training results, the fine-tuning phase leads to a significant degradation in performance metrics.

---

## 🔍 Initial Training Details

- **Model:** YOLOv8 (custom recognition task)
- **Dataset Size:** ~1 million annotated table images
- **Classes Used:**
  - `table_row`
  - `table_column`
  - `table_column_header`
  - ... (others related to table structure)

### ✅ Initial Training Results

| Metric     | Value  |
|------------|--------|
| Precision  | 96%    |
| Recall     | 94%    |
| mAP50      | High   |

The model demonstrated excellent performance on a large-scale dataset for structured table image understanding.

---

## 🔁 Fine-Tuning Details

- **Fine-Tuning Dataset:** 17,000 images
- **Purpose:** Domain-specific refinement using previously trained model
- **Method:** Loaded the trained model weights as a pretrained checkpoint

### ❌ Fine-Tuning Results

![Image](https://github.com/user-attachments/assets/7b44516f-8f0f-44b0-80ae-f3822a69d1e9)

After training with the 17k dataset:

- Precision, Recall, and mAP dropped significantly
- Overall performance was worse than training from scratch on the 17k dataset

---

## 🔬 Troubleshooting Done

- Verified label consistency (class names and indices match)
- Checked annotation formats (correct YOLO format used)
- Ensured data split (train/val) is clean and balanced
- Tried variations in:
  - Learning rate
  - Image size
  - Batch size
- Used both `--resume` and `--weights` strategies for continuation

---

## ❓ Open Questions

1. **Is fine-tuning a YOLOv8 model trained on a large dataset prone to forgetting or instability on smaller datasets?**
2. **Are there recommended techniques (e.g., layer freezing, lower LR, warmup strategy) for preserving performance while fine-tuning on a smaller dataset?**
3. **Could internal normalization statistics (e.g., BN layers) be resetting during fine-tuning, leading to degraded performance?**

---

@glenn-jocher 

## 💡 Example Command Used

```bash
yolo detect train \
  --data data.yaml \
  --img 800\
  --batch 64 \
  --epochs 100\
  --weights path/to/pretrained.pt

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tzktok, thank you for sharing your detailed fine-tuning scenario with Ultralytics 🚀! This is an automated response to assist you while an Ultralytics engineer reviews your issue.

We recommend exploring the [Docs](https://docs.ultralytics.com/) for best practices and troubleshooting, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) guides that may help with common fine-tuning questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) to help us debug, including a small sample of your dataset, training commands, and any logs demonstrating the performance drop.

For custom training or fine-tuning ❓ questions, please continue to share as much detail as possible—dataset formats, sample images, and specifics about your training workflow. Also, review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for guidance on strategies like layer freezing, learning rate adjustments, and batch normalization handling.

Join the Ultralytics community for support and knowledge sharing! For quick chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, see [Discourse](https://community.ultralytics.com/). Or share insights on our [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Upgrade

Make sure you are using the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your report! An Ultralytics engineer will assist you soon.

**Comment by Y-T-G** [comment]
`--weights` is not a valid argument in Ultralytics and the training wouldn't work with it.
https://docs.ultralytics.com/modes/train/#train-settings

Please provide the correct code that was used.

**Comment by tzktok** [comment]
> `--weights` is not a valid argument in Ultralytics and the training wouldn't work with it. https://docs.ultralytics.com/modes/train/#train-settings
> 
> Please provide the correct code that was used.

No.. That's for example below is my code..
```
def model_train(data_yaml_path):
    model = YOLO('pretrained_model.pt')
    data = Path(data_yaml_path)
    results = model.train(data=data, epochs=100, imgsz=800, patience=10, cls=0.25, box=0.05, project="final-table-structure",
                          device=[0, 1], batch=64)
```
@Y-T-G

**Comment by Y-T-G** [comment]
Did you try without adjusting `box` or `cls` weights?

**Comment by tzktok** [comment]
> Did you try without adjusting `box` or `cls` weights?

No..Because we tried same box and cls weights for basemodel..It gives best results.. @Y-T-G
